File: java/rooms/options/Room.java
package rooms.options;

import logic.GameMaster;
import monsters.Hero;
import java.util.ArrayList;
import java.util.List;

public class Room implements GameMasterable {
    private final List<GameMasterable> nextRooms;
    private final GameMaster gameMaster;
    private final RoomType roomType;
//    private int id;

//    public Room(GameMuster gameMuster, RoomType type, int id) {
//        this(gameMuster,type);
//        this.id = id;
//    }

    public Room(GameMaster gameMaster, RoomType type) {
        nextRooms = new ArrayList<>();
        this.gameMaster = gameMaster;
        roomType = type;
    }


    public void addRoom(GameMasterable room) {
        if(nextRooms.size() < 4) nextRooms.add(room);
    }

    @Override
    public void explore(Hero hero) {
    }

    @Override
    public GameMasterable[] getNextRooms() {
        return nextRooms.toArray(new GameMasterable[nextRooms.size()]);
    }

    @Override
    public RoomType getRoomType() {
        return roomType;
    }

    @Override
    public GameMaster getMuster() {
        return gameMaster;
    }
}

File: php/rooms/options/Room.php
<?php

namespace rooms\options;

use logic\GameMaster;
use monsters\Hero;
use interfaces\GameMasterable;
use rooms\RoomType;
use \ArrayList;

class Room implements GameMasterable {
    private $nextRooms;
    private $gameMaster;
    private $roomType;

    public function __construct(GameMaster $gameMaster, RoomType $type) {
        $this->nextRooms = new ArrayList();
        $this->gameMaster = $gameMaster;
        $this->roomType = $type;
    }

    public function addRoom(GameMasterable $room) {
        if(count($this->nextRooms) < 4) {
            $this->nextRooms[] = $room;
        }
    }

    public function explore(Hero $hero) {
    }

    public function getNextRooms() {
        return $this->nextRooms->toArray(new GameMasterable[count($this->nextRooms)]);
    }

    public function getRoomType() {
        return $this->roomType;
    }

    public function getMuster() {
        return $this->gameMaster;
    }
}

?>